calibrators/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,157 @@
+# tag_based_pnp_calibrator
+
+In this tutorial, we present a hands-on tutorial of the `tag_based_pnp_calibrator`. Although we provide a pre-recorded rosbag, the flow of the tutorial is meant to show the user the steps they must perform in their own use cases with live sensors.
+
+General documentation regarding this calibrator can be found [here](../../tag_based_pnp_calibrator/README.md).
+
+## Setup
+
+This tutorial assumes that the user has already built the calibration tools.
+Installation instructions can be found [here](../../../README.md).
+
+## Data preparation
+
+Please download the data (rosbag) from [here](https://drive.google.com/drive/folders/1eO_B_fdBdc8uI5a5LYU4jrVUS0OzRVEq).
+
+The provided rosbag includes four different topics: `/sensing/camera/camera6/camera_info`, `/sensing/camera/camera6/image_rect_color/compressed`, `/sensing/lidar/front_lower/pointcloud_raw`, and `tf_static`.
+
+## Environment preparation
+
+### Overall calibration environment
+
+The required space for calibration depends on the vehicle and sensors used. During calibration, please always make sure that tags are detected by both camera and lidar. The user can always check whether the tags appear in the `rviz` and the `image view` UI. For instance, he can observe the cyan rhombus around the tag in `rviz` if the tag is successfully detected by lidar, as shown in the image below. For the camera, if the tag is visible in the `image view` UI, it should be detectable. So far, in our use cases, the tag is always detected as long as it is completely visible in the image.
```

Comment:
```suggestion
The required space for calibration depends on the vehicle and sensors used. During calibration, please always make sure that tags are detected by both camera and lidar. This can be done by checking whether the tags appear in the `rviz` and the `image view` UI. For instance, a cyan rhombus around the tag in `rviz` represents a successful detection by the lidar, as shown in the image below. For the camera, if the tag is visible in the `image view` UI, it should be detectable. So far, in our use cases, the tag is always detected as long as it is completely visible in the image.
```